packages/admin/src/form/FinalFormRangeSlider.tsx

```diff
@@ -0,0 +1,135 @@
+import { FormControl, Slider, SliderProps, Theme, WithStyles } from "@material-ui/core";
+import { createStyles, withStyles } from "@material-ui/styles";
+import * as React from "react";
+import { FieldRenderProps } from "react-final-form";
+
+import { InputBase } from "./InputBase";
+
+export type CometAdminFinalFormRangeSliderClassKeys =
+    | "root"
+    | "inputsWrapper"
+    | "inputFieldsSeperatorContainer"
+    | "sliderWrapper"
+    | "inputFieldContainer";
+
+const styles = (theme: Theme) => {
+    return createStyles<CometAdminFinalFormRangeSliderClassKeys, any>({
+        root: {
+            padding: "0 20px",
+            width: "100%",
+        },
+        inputsWrapper: {
+            justifyContent: "space-between",
+            marginBottom: "15px",
+            alignItems: "center",
+            display: "flex",
+        },
+        inputFieldsSeperatorContainer: {
+            textAlign: "center",
+            minWidth: "20%",
+        },
+        sliderWrapper: {
+            paddingBottom: "20px",
+        },
+        inputFieldContainer: {
+            textAlign: "center",
+        },
+    });
+};
+
+interface IFinalFormRangeSliderProps extends FieldRenderProps<{ min: number; max: number }, HTMLInputElement> {
+    min: number;
+    max: number;
+    startAdornment?: string | React.ReactElement;
+    endAdornment?: string | React.ReactElement;
+    sliderProps?: Omit<SliderProps, "min" | "max">;
+}
+
+const FinalFormRangeSliderComponent: React.FunctionComponent<WithStyles<typeof styles, true> & IFinalFormRangeSliderProps> = ({
+    classes,
+    min,
+    max,
+    sliderProps,
+    startAdornment,
+    endAdornment,
+    input: { name, onChange, value: fieldValue },
+}) => {
+    const [minInput, setMinInput] = React.useState(fieldValue.min || 0);
```

Comment:
Shouldn't it be possible to just use the field state, and no local state at all?

Reply:
that would require validating the value in the input immediately and not work correctly when you have eg. a min of 10 and want to enter 15 (as 1 < 15)

with the local state the 1 is only in minInput and gets copied into the field value on blur

Reply:
about that question  => i explained him what i wanted to do and after the conversation we came up with a better readable and understandable solution for it :)

Reply:
Yes, I didn't understand the "issue" by just reading the code, but @MFSepplive explained it to me and this is perfectly fine 👍🏼

Reply:
there always rings an alarm when you see useState in a form field :)